integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,529 @@
+			Cluster:    p.cluster,
+			Username:   p.username,
+			RouteToDatabase: tlsca.RouteToDatabase{
+				ServiceName: p.cluster + "-postgres",
+				Protocol:    defaults.ProtocolPostgres,
+				Username:    "postgres",
+				Database:    "test",
+			},
+		})
+		require.NoError(t, err)
+
+		result, err := connClient.Exec(context.Background(), "select 1").ReadAll()
+		require.NoError(t, err)
+		require.Equal(t, []*pgconn.Result{postgres.TestQueryResponse}, result)
+		require.Equal(t, uint32(i+1), p.postgresDB.QueryCount())
+
+		err = connClient.Close(context.Background())
+		require.NoError(t, err)
+	}
+}
+
+// makeLoadBalancer bootsraps a new load balancer for proxy instances.
+func (p *proxyTunnelStrategy) makeLoadBalancer(t *testing.T) {
+	if p.lb != nil {
+		require.Fail(t, "load balancer already initialized")
+	}
+
+	lbAddr := utils.MustParseAddr(net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt())))
+	lb, err := utils.NewLoadBalancer(context.Background(), *lbAddr)
+	require.NoError(t, err)
+
+	require.NoError(t, lb.Listen())
+	go lb.Serve()
+
+	p.lb = lb
+}
+
+// makeAuth bootsraps a new teleport auth instance.
+func (p *proxyTunnelStrategy) makeAuth(t *testing.T) {
+	if p.auth != nil {
+		require.Fail(t, "auth already initialized")
+	}
+
+	privateKey, publicKey, err := testauthority.New().GenerateKeyPair()
+	require.NoError(t, err)
+
+	auth := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		Priv:        privateKey,
+		Pub:         publicKey,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	auth.AddUser(p.username, []string{p.username})
+
+	conf := service.MakeDefaultConfig()
+	conf.DataDir = t.TempDir()
+	conf.Auth.Enabled = true
+	conf.Auth.NetworkingConfig.SetTunnelStrategy(p.strategy)
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+
+	require.NoError(t, auth.CreateEx(t, nil, conf))
+	require.NoError(t, auth.Start())
+
+	p.auth = auth
+}
+
+// makeProxy bootstraps a new teleport proxy instance.
+// It's public address points to a load balancer.
+func (p *proxyTunnelStrategy) makeProxy(t *testing.T) {
+	proxy := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	authAddr := utils.MustParseAddr(net.JoinHostPort(p.auth.Hostname, p.auth.GetPortAuth()))
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, *authAddr)
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.SSH.Enabled = false
+
+	conf.Proxy.Enabled = true
+	conf.Proxy.WebAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PeerAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PublicAddrs = append(conf.Proxy.PublicAddrs, utils.FromAddr(p.lb.Addr()))
+	conf.Proxy.DisableWebInterface = true
+
+	require.NoError(t, proxy.CreateEx(t, nil, conf))
```

Comment:
This call to `CreateEx` will redefine `conf.Proxy.WebAddr` (and the other listeners) and will add the `proxy` `TeleInstance`'s intended internal auth server's address to `conf.AuthServers`.

Also, AIUI nothing ever enables multiplex mode - not sure if we care about that here, but it's weird that we're running a single loadbalancer just for the web port while everything else is split between proxies.